basal.py

```diff
@@ -0,0 +1,69 @@
+import random
+import common_fields 
+import settings
+from datetime import datetime, timedelta
+from pytz import timezone
+import tools
+
+def scheduled_basal(start_time, params):
+    basal_data = []  
+    access_settings = settings.settings(start_time, params)[0]
+    next_time = int(start_time.strftime('%s')) #in seconds
+    seconds_to_add = params["num_days"] * 24 * 60 * 60
+    end_date = start_time + timedelta(seconds=seconds_to_add)
+    end_time = int(end_date.strftime('%s'))
+    while next_time < end_time:
+        basal_entry = {}
+        basal_entry = common_fields.add_common_fields('basal', basal_entry, next_time, params)
+        basal_entry["deliveryType"] = "scheduled"   
+        schedule = access_settings["basalSchedules"]["standard"] 
+        basal_entry["rate"], start, initial_start, end = tools.get_rate_from_settings(schedule, basal_entry["deviceTime"] , "basalSchedules")
+        duration = (end - start) / 1000 #in seconds
+        zone = timezone(params['zone'])
+        start_date, end_date = datetime.fromtimestamp(next_time), datetime.fromtimestamp(next_time + duration)
+        localized_start, localized_end = zone.localize(start_date), zone.localize(end_date)
+        start_offset, end_offset = tools.get_offset(params['zone'], start_date), tools.get_offset(params['zone'], end_date)
+        if start_offset != end_offset:
+            diff = end_offset - start_offset
+            localized_end = localized_end - timedelta(minutes=diff)
+        elapsed_seconds = (localized_end - localized_start).total_seconds()
+        if next_time == int(start_time.strftime('%s')):
+            basal_entry["duration"] = end - initial_start
+        else:
+            basal_entry["duration"] = int(elapsed_seconds * 1000)
+        basal_entry["scheduleName"] = "standard"
+       
+        if randomize_temp_basal(): #create temp basal if true
+            basal_data.append(temp_basal(basal_entry, next_time, params))
+        
+        if randomize_temp_basal():
+            basal_data.append(suspended_basal(basal_entry, next_time, params))    
+
+        next_time += basal_entry["duration"] / 1000
+        basal_data.append(basal_entry)
+    return basal_data
+
+def temp_basal(scheduled_basal, timestamp, params):
+    basal_entry = {}
+    basal_entry = common_fields.add_common_fields('basal', basal_entry, timestamp, params)
+    basal_entry["deliveryType"] = "temp"
+    basal_entry["duration"] = scheduled_basal["duration"] 
+    basal_entry["percent"] = random.randrange(0, 80, 5) / 100
```

Comment:
Temp basals can be *increases* too, so the range should be up to maybe 200% (but make sure you exclude values of 100% somehow).